### PR TITLE
Fix dependency of STAR geometry libraries on generated source files

### LINKED ISSUE #45: StarVMC/StarGeometry doesn't compile on main branch
### Steps to reproduce

On RACF run:
```bash
git clone https://github.com/star-bnl/star-sw.git
cd star-sw
git checkout f7935bbbca1c9e9c69931bb5167ca9767be72807
env -i \
  PATH=$PWD/mgr:$PATH \
  QTDIR=/usr/lib64/qt-3.3 QTINC=/usr/lib64/qt-3.3/include \
  QTLIB=/usr/lib64/qt-3.3/lib CERN_LEVEL=pro \
  CERN=/cern CERN_ROOT=/cern/pro \
  ROOTSYS=/afs/rhic.bnl.gov/star/ROOT/5.34.38/.sl73_gcc485/rootdeb \
  ROOT=/afs/rhic.bnl.gov/star/ROOT \
  ROOT_LEVEL=5.34.38 \
  STAR_SYS=x8664_sl7 \
  STAR_HOST_SYS=sl73_gcc485 \
  OPTSTAR=/opt/star/sl73_gcc485 \
  XOPTSTAR=/afs/rhic.bnl.gov/opt/star/sl73_gcc485 \
  STAR_ROOT=/afs/rhic.bnl.gov/star \
  STAR=/nowhere \
  sh -c '. $ROOTSYS/bin/thisroot.sh; exec cons %StShadowMaker'
```

Observe:
```
Install StarVMC/StarGeometry/BtofConfig.cxx as .sl73_gcc485/obj/StarVMC/StarGeometry/BtofConfig.cxx
g++ -m32 -fPIC -pipe -Wall -Woverloaded-virtual -std=c++0x -Wno-long-long -pthread -g -Dsl73_gcc485 -D__ROOT__ -DNEW_DAQ_READER -I. -IStRoot -I.sl73_gcc485/include -I/afs/rhic.bnl.gov/star/ROOT/5.34.38/.sl73_gcc485/rootdeb/include -c .sl73_gcc485/obj/StarVMC/StarGeometry/BtofConfig.cxx -o .sl73_gcc485/obj/StarVMC/StarGeometry/BtofConfig.o
In file included from .sl73_gcc485/obj/StarVMC/StarGeometry/BtofConfig.cxx:1:0:
./StarVMC/StarGeometry/BtofConfig.h:3:45: fatal error: StarVMC/StarGeometry/BtofGeo1.h: No such file or directory # include "StarVMC/StarGeometry/BtofGeo1.h"                                              ^
compilation terminated.
cons: *** [.sl73_gcc485/obj/StarVMC/StarGeometry/BtofConfig.o] Error 1
cons: errors constructing .sl73_gcc485/obj/StarVMC/StarGeometry/BtofConfig.o
```

### Expected result

Everything compiles without errors

cc @klendathu2k

### PR DESCRIPTION
**Pull request by plexoos**
This is yet another attempt to resolve #45 
Closes: #92 #146

### CODE REVIEW

**Review by klendathu2k — COMMENTED** [review]
I expect that this will all work, but I should check for side effects... before approving...

**Comment by klendathu2k** [comment]
The source file StarVMC/xgeometry/xgeometry.age is installed into the build area, but it is never compiled.  This file is _required_ to steer the creation of the geometry.

$ ls .sl73_x8664_gcc485/obj/StarVMC/xgeometry/xgeometry.age
.sl73_x8664_gcc485/obj/StarVMC/xgeometry/xgeometry.age
$ ls .sl73_x8664_gcc485/obj/StarVMC/xgeometry/xgeometry.F
ls: cannot access .sl73_x8664_gcc485/obj/StarVMC/xgeometry/xgeometry.F: No such file or directory
$ ls .sl73_x8664_gcc485/obj/StarVMC/xgeometry/xgeometry.o
ls: cannot access .sl73_x8664_gcc485/obj/StarVMC/xgeometry/xgeometry.o: No such file or directory

**Comment by plexoos** [comment]
Hm, that's strange. I do see these files (xgeometry.age/.F/.o) in the build log starting with this line https://github.com/star-bnl/star-sw/pull/148/checks?check_run_id=3631646347#step:3:28851

You can search for 'xgeometry.' to verify

To me it looks like the object file gets into the xgeometry.so library. I wonder if your $STAR is set to /afs/... and somehow interferes with your local source. Try setting $STAR to your local checked out star-sw directory

**Comment by klendathu2k** [comment]
Confirmed.  `setenv STAR .` results in a correct build of the geometry.  Moving this code will not impact building production libraries.  Users may have to work around if AFS is confusing, but I believe that `cons -noR` may work there.

**Review by klendathu2k — APPROVED** [review]
Approved.  If users run into build issues, try `cons -noR`

File: mgr/Conscript-standard [hunk 1]
```diff
@@ -376,34 +378,25 @@ if ( $pkg !~ /^sim$/ && $pkg !~ /^gen$/ ) {
...
+			    Depends $env ["$OBJ/$agX", "$OBJ/$agH", "$OBJ/$agE"], ("#$xml", "#$agml_cmd");
 			    # Depends $env --> some rule to build extra dict on the fly
 
-			    Command $env ["#".$agX], ("#$xml"), qq($agml_cmd --file=%1 --module=$agM --export=AgROOT --path=%>:d);
-			    Command $env ["#".$agE], ("#$xml"), qq($agml_cmd --file=%1 --module=$agM --export=Mortran > %>);
+			    Command $env ["$OBJ/$agX", "$OBJ/$agH"], ("#$xml"), qq($agml_cmd --file=%1 --module=$agM --export=AgROOT --path=%>:d);
+			    Command $env ["$OBJ/$agE"], ("#$xml"), qq($agml_cmd --file=%1 --module=$agM --export=Mortran > %>);
 
 			    # Install local file (#) as global
 			    # The .h file can be installed as-is
```
Comment (veprbl) [inline mgr/Conscript-standard]: This comment is now outdated
```suggestion
```

File: mgr/Conscript-standard [hunk 2]
```diff
@@ -288,26 +288,28 @@ if ( $pkg !~ /^sim$/ && $pkg !~ /^gen$/ ) {
...
+		    $trgt = "$OBJ/$Dir/$agM.cxx";
+		    $hfl  = "$OBJ/$Dir/$agM.h";
 		}
 	    }
 	    # push but only if not present - if present, cons will find
 	    # them so there is no need for a predictive target rule
-	    push(@src, $trgt ) if ($trgt ne "" );
-	    push(@inc, $hfl)   if ($hfl  ne "" );
+	    push(@src, $trgt);
+	    push(@inc, $hfl);
```
Comment (veprbl) [inline mgr/Conscript-standard]: Are we sure about removing the condition? For example, `$hfl` will be empty for `$Dir = xgeometry`

**Review by veprbl — APPROVED** [review]
Makes sense!
And I love how it produces output files to $OBJ, this is :100:

File: mgr/Conscript-standard (hunk 2 above)
Comment (plexoos) [reply mgr/Conscript-standard]: Sure, let's keep it

**Comment by plexoos** [comment]
Two approvals, checks passed... Last call for any other comments

A slight off-topic but somewhat related.  As a good side effect of this I realized that rootcling in ROOT6 is still expecting to find header files when loading libraries with `gSystem->Load()` so, having the geometry .h files in our $INCLUDE helps. I am not sure though why it is not enough for rootcling to see the pcm files which we put next to the libraries... Perhaps Victor @perevbnlgov could comment?

**Comment by veprbl** [comment]
Last time I've checked, the paths to pcm files were hardcoded into the libraries, so they are not relocable. But we act as if the were:

https://github.com/star-bnl/star-sw/blob/bc1212b65bb422399a2eb1fc9dc3aa47a474911e/mgr/Conscript-standard#L1131-L1132

**Comment by plexoos** [comment]
We don't really relocate the pcm files. The Install directive in Cons creates a link to the original file in $OBJ.
My note is related to the header files which ROOT still tries to find when a library is loaded. My expectation is that with pcm files the headers are not needed but my expectation can be wrong.

**Pull request merged**